### PR TITLE
feat: modify job_config & update mapping table

### PR DESCRIPTION
**Pull request by owem0410**
1. the load job to append the data to the table and allowing nullable field addition.
2. update mapping table for  2021 kktix data.

## Types of changes
- **New feature**

## Checklist:
- [ ] Add test cases to all the changes you introduce
- [ ] Run `poetry run pytest` locally to ensure all linter checks pass
- [ ] Update the documentation if necessary

### CODE REVIEW

**Comment by tai271828** [comment]
1. CI does not pass
2. Would you mind elaborating a bit more why survey_cake relates to kktix?? I must have a gap of any updated workflow.

**Comment by david30907d** [comment]
@owem0410 is it WIP or ready to review? I guess it's still WIP 😂

**Comment by owem0410** [comment]
> 2. Would you mind elaborating a bit more why survey_cake relates to kktix?? I must have a gap of any updated workflow.
The pr follow the pitch "KKTIX, SurveyCake 的上傳問卷要能合併個年度的資料進同張表". So I modify the survey_cake_csv_uploader.py ,too.

**Comment by owem0410** [comment]
> @owem0410 is it WIP or ready to review? I guess it's still WIP 😂
I'm sure the status is "ready to review"......
Let me check why the CI does not pass....

**Comment by david30907d** [comment]
I see, @owem0410 if you get a chance, would you assign a reviewer or c.c. others next time?
Otherwise I might don't know it's ready to review 😂 
thx 🙏

**Comment by david30907d** [comment]
@owem0410 seems to me that you didn't use the `npm run commit`. In other words, no linter/formatter was applied to your commit 😅

File: contrib/upload-kktix-ticket-csv-to-bigquery.py [hunk 1]
```diff
@@ -162,6 +162,11 @@
     "是否願意提供 Email 給贊助商": "email_to_sponsor",
     # from 2018 reformatted column names
     "size_of_tshirt_t": "size_of_tshirt",
+    # from 2021 reformatted column names
```
Comment (david30907d) [inline contrib/upload-kktix-ticket-csv-to-bigquery.py:165]: thx for the comment~

File: contrib/survey_cake/udfs/survey_cake_csv_uploader.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ def _upload_2_bigquery(self, file_path, table_id):
             autodetect=True,
             allow_quoted_newlines=True,
             write_disposition="WRITE_APPEND",
+            schema_update_options="ALLOW_FIELD_ADDITION",
```
Comment (david30907d) [inline contrib/survey_cake/udfs/survey_cake_csv_uploader.py:90]: just realized we need this to make our table being `NULLABLE`
👍 good to know

File: contrib/upload-kktix-ticket-csv-to-bigquery.py [hunk 3]
```diff
@@ -190,9 +200,13 @@ def upload_dataframe_to_bigquery(
 
     dataset_ref = bigquery.dataset.DatasetReference(project_id, dataset_name)
     table_ref = bigquery.table.TableReference(dataset_ref, table_name)
-
+    job_config = bigquery.LoadJobConfig()
+    job_config.write_disposition = bigquery.WriteDisposition.WRITE_APPEND
```
Comment (david30907d) [inline contrib/upload-kktix-ticket-csv-to-bigquery.py:204]: 👍

**Review by david30907d — APPROVED** [review]
utACK

**Comment by owem0410** [comment]
> @owem0410 seems to me that you didn't use the `npm run commit`. In other words, no linter/formatter was applied to your commit 😅

OK,Let me try.

**Pull request merged**